gc.c

```diff
@@ -362,7 +362,8 @@ sexp sexp_finalize (sexp ctx) {
       for (r=q->next; r && ((char*)r<(char*)p); q=r, r=r->next)
         ;
       if ((char*)r == (char*)p) { /* this is a free block, skip it */
-        p = (sexp) (((char*)p) + r->size);
+        // ISSUE_296 - Access to 'size' dereference NULL 'r'
+        p = (sexp) (((char*)p) + (r ? r->size : 0));
```

Comment:
Whatever the static analyzer was complaining about it had 18 arrows to describe the situation.  Changed the code as follows to silence the warning - should have no effect as case is not possible.

```c
        p = (sexp) (((char*)p) + (r ? r->size : 0));
```